app/code/Magento/Sitemap/Model/ItemProvider/Composite.php

```diff
@@ -32,7 +32,7 @@ public function getItems($storeId)
     {
         $items = [];
 
-        foreach ($this->itemResolvers as $resolver) {
+        foreach ($this->itemProviders as $resolver) {
             foreach ($resolver->getItems($storeId) as $item) {
                 $items[] = $item;
```

Comment:
@Koc goal of this pull request was to improve way to customise for 3rd party extensions to add how items are added without need to extend or override sitemap class itself. I haven't changed way how data is extracted from database itself and passed around. This is way I came up with while working on it in my free personal time and it was accepted as good enough solution. If you are able to propose more efficient way feel free to create and issue with your ideas as a way to start discussion or even better PR with an implementation.